src/App.tsx

```diff
@@ -1,41 +1,70 @@
-import React from 'react';
+import React, { useState, useEffect } from 'react';
 import './App.scss';
 import './styles/general.scss';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
+import { getUserPosts } from './api/posts';
+import { getUsers } from './api/users';
 
-const App: React.FC = () => (
-  <div className="App">
-    <header className="App__header">
-      <label>
-        Select a user: &nbsp;
-
-        <select className="App__user-selector">
-          <option value="0">All users</option>
-          <option value="1">Leanne Graham</option>
-          <option value="2">Ervin Howell</option>
-          <option value="3">Clementine Bauch</option>
-          <option value="4">Patricia Lebsack</option>
-          <option value="5">Chelsey Dietrich</option>
-          <option value="6">Mrs. Dennis Schulist</option>
-          <option value="7">Kurtis Weissnat</option>
-          <option value="8">Nicholas Runolfsdottir V</option>
-          <option value="9">Glenna Reichert</option>
-          <option value="10">Leanne Graham</option>
-        </select>
-      </label>
-    </header>
-
-    <main className="App__main">
-      <div className="App__sidebar">
-        <PostsList />
-      </div>
-
-      <div className="App__content">
-        <PostDetails />
-      </div>
-    </main>
-  </div>
-);
+const App: React.FC = () => {
+  const [selectedUser, selectUser] = useState(0);
+  const [posts, setPosts] = useState([] as Post[]);
+  const [selectedPostId, selectPost] = useState(0);
+  const [users, setUsers] = useState([] as User[]);
+
+  useEffect(() => {
+    getUsers()
+      .then(serverUsers => setUsers(serverUsers));
+  }, []);
+
+  useEffect(() => {
+    getUserPosts(selectedUser)
+      .then(serverPosts => setPosts(serverPosts));
+  }, [selectedUser]);
+
+  return (
+    <div className="App">
+      <header className="App__header">
+        <label>
+          Select a user: &nbsp;
+
+          <select
+            className="App__user-selector"
+            onChange={(event) => selectUser(+event.target.value)}
+            value={selectedUser}
+          >
+            <option value="0">All users</option>
+            {
+              users.map(user => (
+                <option
+                  value={user.id}
+                  key={user.id}
+                >
+                  {user.name}
+                </option>
+              ))
+            }
+          </select>
+        </label>
+      </header>
+
+      <main className="App__main">
+        <div className="App__sidebar">
+          <PostsList
+            posts={posts}
+            openPost={selectPost}
+            selectedPostId={selectedPostId}
+          />
+        </div>
+
+        <div className="App__content">
+          {!!selectedPostId && (
```

Comment:
```suggestion
          {selectedPostId && (
```

Reply:
If I wrote `selectedPostId` without `!!`, react would render 0 instead of user card (in case no user is selected)